ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-filter/index.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import styled from '../../styles/styled-components'
+
+import { Button, buttonTypeEnum } from '../button'
+
+import { MarionetteRegionContainer } from '../../container/marionette-region-container'
+
+const FilterBuilderView = require('../../../component/filter-builder/filter-builder.view')
+
+type Props = {
+  removeFilter: () => void
+  saveFilter: (transformToCql: () => any) => void
+  hasFilter: Boolean
+  resultFilter?: {}
+  isList?: Boolean
+}
+
+type State = {
+  filterCqlCallback: () => any
+  resultFilter?: {}
+}
+
+type BuilderView = {
+  turnOnEditing: () => void
+  turnOffNesting: () => void
+  transformToCql: () => string
+}
+
+const Root = styled.div`
+  display: block;
+  padding: ${props => props.theme.minimumSpacing};
+`
+
+const EditorFooter = styled.div`
+  padding-top: 30px;
+  white-space: nowrap;
+`
+
+const SaveButtonStyle = (props: Props) =>
+  ({
+    display: 'inline-block',
+    width: (props.hasFilter && '50%') || '100%',
+  } as React.CSSProperties)
+
+const RemoveButtonStyle = {
+  display: 'inline-block',
+  width: '49%',
+  marginRight: '1%',
+} as React.CSSProperties
+
+class BuilderViewRegionContainer extends MarionetteRegionContainer {
+  onceInDOM(callback: () => void) {
```

Comment:
@andrewkfiedler This is what I came up with in order to be able to call methods on the child view while leaving the container alone.

Reply:
Since we know that `onBeforeShow` is the moment that it's safe to do what we need, we can create a slightly modified version of the FilterBuilderView that will do exactly what we want.  
```
const CustomFilterBuilderView = FilterBuilderView.extend({
  onBeforeShow() {
    FilterBuilderView.prototype.onBeforeShow.call(this)
    this.turnOnEditing()
    this.turnOffNesting()
  },
})
```
This keeps the logic located as close as possible to what it depends on and avoids any uncertainty as to whether or not child views will be ready.  When onBeforeShow is called, the view will be constructed and then editing and nesting will be turned off.